services/libs/data-access-layer/src/members/segments.ts

```diff
@@ -61,6 +62,24 @@ export async function fetchManyMemberSegments(
   )
 }
 
+export async function fetchSegmentsByIds(
+  qe: QueryExecutor,
+  ids: string[],
+  fields = '*',
+): Promise<ISegment[]> {
+  const formattedIds = ids.join(', ')
+  return qe.select(
+    `SELECT ${fields}
+    FROM "segments"
+    WHERE id IN ($(ids))
+        `,
+    {
+      ids: formattedIds,
+      fields,
+    },
+  )
+}
+
```

Comment:
You don't need this, we already have `fetchManySegments()` in `services/libs/data-access-layer/src/segments/index.ts`